example/provider-extensions/registry-seed/registry.yaml

```diff
@@ -0,0 +1,96 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: registry
+  labels:
+    app: registry     
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: registry
+  namespace: registry
+  labels:
+    app: registry
+spec:
+  type: ClusterIP
+  selector:
+    app: registry
+  ports:
+  - port: 5000
+    targetPort: registry
+    protocol: TCP
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: registry
+  namespace: registry
+  labels:
+    app: registry
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: registry
+  template:
+    metadata:
+      labels:
+        app: registry
+    spec:
+      automountServiceAccountToken: false
+      containers:
+      - name: registry
+        image: registry:2
+        imagePullPolicy: IfNotPresent
+        ports:
+        - name: registry
+          containerPort: 5000
+        volumeMounts:
+        - name: cache
+          mountPath: /var/lib/registry
+      volumes:
+      - name: cache
+        emptyDir: {}        
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: registry-ingress
+  namespace: registry
+  annotations:
+    # TLS cert by managed gardener
+    cert.gardener.cloud/purpose: managed
+    # DNS managed by gardener
+    dns.gardener.cloud/class: garden
+    dns.gardener.cloud/ttl: "600"
+    dns.gardener.cloud/dnsnames: $HOST
```

Comment:
Same here. I'm not too happy about the assumption that the seed cluster is a shoot from somewhere else. This will not work for the vast majority of outside-SAP collaborators.

Reply:
My assumption was that people contribute to Gardener because they use Gardener 😄 

Things like DNS and TLS certificates are very cumbersome to maintain for the different hyperscalers, but very easy when you assume, that there is a Gardener cluster. WDYT about two modes, a "Gardener mode" and a "manual mode"?

Reply:
Yeah, I'm fine with optimising the flow if we detect that the seed is a shoot from another Gardener (and then use the `shoot-info` configmap as it's done now). However, in general we also need the other mode so that contributors can set this up for non-shoot seeds. Maybe some "config" file that one can put into the `./dev` directory which contains the necessary information that you would otherwise read from the `shoot-info` configmap.

Reply:
See before.
On non-Gardener clusters the DNS entries for the container registry and relay server have to be created manually. There are too many different options for an automation.

Reply:
I kind of agree with @rfranzke because we don't only rely on the shoot but also that the shoot has DNS service. We should at least mention this specific assumption in the docs and elaborate on the manual alternative.

Reply:
The docs have not been adapted, but the [configure-seed.sh](https://github.com/oliver-goetz/gardener/blob/f62ecfd06f51b7fbd7378e293d5bcbc49b95326e/example/provider-extensions/seed/configure-seed.sh#L146-L156) script was already pointing people into the right direction.